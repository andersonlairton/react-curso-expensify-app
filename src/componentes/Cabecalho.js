import React from 'react';
import {BrowserRouter,Route,Switch,Link,NavLink} from 'react-router-dom';

const Cabecalho=()=>
(
    <div>
        <h1>Exepensivo App</h1>
        <NavLink to="/" activeClassName="is-active"/* chamando elementos da classe */ exact={true}>Home</NavLink>
        <NavLink to="/help" activeClassName="is-active">Ajuda</NavLink>
        <NavLink to="/edit" activeClassName="is-active">Editar</NavLink>
        <NavLink to="/create" activeClassName="is-active">Create page</NavLink>
    </div>
);

export default Cabecalho;