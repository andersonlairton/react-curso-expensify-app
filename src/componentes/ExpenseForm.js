import React from 'react'
import moment from 'moment';

import { SingleDatePicker } from 'react-dates';
 //import 'react-dates/lib/css/_datepicker.css';

//const now = moment();//CONFIGURANDO DATA
//console.log(now.format('MMM Do,YYYY'));
export default class ExpenseForm extends React.Component {
    constructor (props)
    {
        super(props);
        this.state =
        {
            descricao: props.expense?props.expense.descricao:'',
            note: props.expense?props.expense.note:'',
            amount: props.expense?(props.expense.amount/100).toString():'',
            createAt: props.expense? moment(props.expense.createAt):moment(),
            calendarFocused: false,
            error:''
        };
    }
    
    onDescriptionChange = (e) => {
        const descricao = e.target.value;
        this.setState(() => ({ descricao }));
    }
    onNoteCharge = (n) => {
        const note = n.target.value;
        this.setState(() => ({ note }));
    }
    onAmountChange = (a) => {
        const amount = a.target.value;
        if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
            this.setState(() => ({ amount }));
        }
    }
    onDateChange = (createAt) => {
        if (createAt) {
            this.setState(() => ({ createAt }));
        }
    }

    onFocusChange = ({ focused }) => {
        this.setState(() => ({ calendarFocused: focused }));
    }
/* 
    enviarForm=(e)=> {
        e.preventDefault();
        if(!this.state.descricao||!this.state.amount)
        {
            this.setState(()=>({error:'digite os dados para enviar'}));
            console.log('digite dados para enviar');

        }
        else
        {
            this.setState(()=>({error:''}));
           // console.log('enviado');
            //enviado dados
            this.props.onSubmit(
                {
                    descricao:this.state.descricao,
                    amount:parseFloat(this.state.amount,10)*100,
                    createAt:this.state.createAt.valueOf(),
                    note:this.setState.note
                });
        }

    }*/
   enviarForm = (e) => {
        e.preventDefault();
    
        if (!this.state.descricao || !this.state.amount) 
        {
          this.setState(() => ({ error: 'digite os dados para enviar' }));
        } else 
        {
          this.setState(() => ({ error: '' }));
          this.props.onSubmit(
            {
            descricao: this.state.descricao,
            amount: parseFloat(this.state.amount, 10) * 100,
            createAt:this.state.createAt.valueOf(),
            note: this.state.note
            }
            );
        }
      };

      
    render() {
        return (
            <div>
            {this.state.error &&<p>{this.state.error}</p>}
                <fieldset>
                    Adicionar despensa
                </fieldset>
                <form onSubmit={this.enviarForm}>
                    <input
                        type="text"
                        placeholder="descrição"
                        autoFocus
                        value={this.state.descricao}
                        onChange={this.onDescriptionChange}
                    />

                    <input type="text"
                        placeholder="amount"
                        value={this.state.amount}
                        onChange={this.onAmountChange}

                    />

                    <SingleDatePicker//configurando campo data
                        onDateChange={this.onDateChange}
                        focused={this.state.calendarFocused}
                        onFocusChange={this.onFocusChange}
                        numberOfMonths={1}
                        isOutsideRange={() => false}
                    />

                    <textarea
                        placeholder="adicionar nota é opcional"
                        value={this.state.note}
                        onChange={this.onNoteCharge}
                    />
                    <button>Adicionar despensa</button>
                </form>
            </div>
        )
    }
}
