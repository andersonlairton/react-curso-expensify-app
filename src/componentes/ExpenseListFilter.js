/*import React from 'react';
import { connect } from 'react-redux';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters';
import { DateRangePicker } from 'react-dates';

class ExpenseListFilters extends React.Component {
    state = {
            calendarFocused: null
        };
    onDateChange = ({ startDate, endDate }) => {
        this.props.dispatch(setStartDate(startDate));
        this.props.dispatch(setEndDate(endDate));
    };
    onFocusChange=(calendarFocused)=> {
        this.setState(()=>{calendarFocused});
    }
    render() {
        return (
            <div>

                <input type='text' value={this.props.filters.text} onChange={(e) => {
                    //pesquisando os valores digitados no textbox
                    this.props.dispatch(setTextFilter(e.target.value));
                    console.log(e.target.value)
                }} />
                <select/*filtragem pre-definida no combobox */
  /*                 value={this.props.filters.sortBy}
                    onChange={(e) => {
                        if (e.target.value === 'date') {
                            this.props.dispatch(sortByDate());
                        }
                        else if (e.target === 'amount') {
                            this.props.dispatch(sortByAmount());

                        }
                    }
                    }
                >
                    <option value="date">date</option>
                    <option value="amount">amount</option>
                </select>
                <DateRangePicker
                    startDate={this.props.filters.startDate}
                    endDate={this.props.filters.endDate}
                    onDatesChange={this.onDateChange}
                    focusedInput={this.state.calendarFocused}
                    onFocusChange={this.onFocusChange}
                    showClearDates={true}
                    numberOfMonths={1}
                    isOutsideRange={()=>false}


                    
                />
            </div>
        );
    }
}

const mapStateProps = (state) => {
    return {
        filters: state.filters
    }
}
export default connect(mapStateProps)(ExpenseListFilters); */

import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters';

export class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null
  };
  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  };
  onFocusChange = (calendarFocused) => {
    this.setState(() => ({ calendarFocused }));
  }
  onTextChange=(e) => {
    this.props.setTextFilter(e.target.value);
  };
  onSortChange= (e) => {
    if (e.target.value === 'date') {
      this.props.sortByDate();
    } else if (e.target.value === 'amount') {
      this.props.sortByAmount();
    }
  }
  render() {
    return (
      <div>
        <input
          type="text"
          value={this.props.filters.text}
          onChange={this.onTextChange}
        />
        <select
          value={this.props.filters.sortBy}
          onChange={this.onSortChange}
        >
          <option value="date">Date</option>
          <option value="amount">Amount</option>
        </select>
        <DateRangePicker
          startDate={this.props.filters.startDate}
          endDate={this.props.filters.endDate}
          onDatesChange={this.onDatesChange}
          focusedInput={this.state.calendarFocused}
          onFocusChange={this.onFocusChange}
          showClearDates={true}
          numberOfMonths={1}
          isOutsideRange={() => false}
        />
      </div>
    );
  }
};

const mapStateToProps = (state) => ({
  
    filters: state.filters
  
});
const mapDispatchProps=(dispatch)=>(
  {
    setTextFilter:(text)=>dispatch(setTextFilter(text)),
    sortByDate:()=>dispatch(sortByDate()),
    sortByAmount:()=>dispatch(sortByAmount()),
    setStartDate:(startDate)=>dispatch(setStartDate(startDate)),
    setEndDate:(endDate)=>dispatch(setEndDate(endDate))

  });

export default connect(mapStateToProps,mapDispatchProps)(ExpenseListFilters);

