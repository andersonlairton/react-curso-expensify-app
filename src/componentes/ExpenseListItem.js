//listando dados da matriz
import React from 'react';
import {connect}from 'react-redux';
import {removeExpense} from '../actions/expenses';
import {Link} from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

const ExpenseListItem = ({dispatch,id,descricao,amount,createAt})=>
(
   
    <div>
    
    <Link to={`/edit/${id}`} /*criando uma pagina de edição*/>
    <h3>{descricao}</h3>
    </Link>
        
        <p>-{numeral(amount/100).format('$0,0.00')}
        -
        {moment(createAt).format('Do / MMMM / YYYY')}</p>
        
    </div>
);

export default ExpenseListItem;

