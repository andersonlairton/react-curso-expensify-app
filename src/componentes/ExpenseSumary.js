import React from 'react';
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expenses_total';

export const ExpensesSumary = ({ ExpenseContador,expenseCount, ExpenseTotal }) => {
 
    const Despesa = ExpenseContador === 1 ? 'expense' : 'expenses'  ;
    
    const formatoValorTotal = numeral(ExpenseTotal / 100).format('$0,0.00');

    return (
        <div>
            <h1>Veja o valor total das {ExpenseContador} {Despesa} totais ,com o valor de {formatoValorTotal}</h1>

        </div>
    );
}

//adicionado valores as stores(lojas),para ser acessado por outras paginas

const mapStateToprops = (state) => {
    //configurando seletores
    const visibleExpenses = selectExpenses(state.expenses, state.filters);

    return {
        ExpenseContador: visibleExpenses.length,
        
       
        ExpenseTotal: selectExpensesTotal(visibleExpenses)
    };
};
//exportando tudo 
export default connect(mapStateToprops)(ExpensesSumary);