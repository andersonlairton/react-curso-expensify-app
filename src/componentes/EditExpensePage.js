import React from 'react';
import { connect } from 'react-redux'
import ExpenseForm from './ExpenseForm';
import { editExpense, removeExpense } from '../actions/expenses';

/*const EditExpensePage=(props)=>{

    console.log(props);
    return(
    <div>
    pagina de edição do id {props.match.params.id}
    </div>)
};*/

export class EditExpensePage extends React.Component{//definindo elemento de classe
  
  onSubmit = (expense) => {
    this.props.editExpense(this.props.expense.id, expense);//editanado
    this.props.history.push('/');//redirecionando
  }
  onRemove = () => {//removendo itens
    this.props.removeExpense({ id: this.props.expense.id });
    this.props.history.push('/');
  }
  render() {
    return(
    <div>
      <ExpenseForm
        expense={this.props.expense}
        onSubmit={this.onSubmit}
      />
      <button onClick={this.onRemove}>Remove</button>
    </div>
    );
  };
}

const mapStateToProps = (state, props) => (
  {
  
    expense: state.expenses.find((expense) => expense.id === props.match.params.id)
  }
);


const mapDispachToProps = (dispatch,props) => (
  {
    editExpense:(id,expense)=>dispatch(editExpense(id, expense)),
    removeExpense:(data)=>dispatch(removeExpense(data))

  }
);

export default connect(mapStateToProps,mapDispachToProps)(EditExpensePage);