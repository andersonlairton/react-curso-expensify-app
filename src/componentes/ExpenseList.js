import React from 'react';
import { connect } from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import selectExpenses from '../selectors/expenses';

export const ExpenseList = (props) =>
    (
        <div>
            <h1>Componente da pagina ExpenseList</h1>

            {props.expenses.length === 0 ? (
                <p>no expenses</p>
            ) : (


                    //  {<p>filtro:{props.filters.text}</p> }

                    props.expenses.map((expense) => {
                        return <ExpenseListItem key={expense.id} {...expense} />;
                    }
                    )
                    )
            }




        </div>
    );
const mapStateToProps = (state) => {
    return {
        //filtrando dados conforme filtros da pagina selectExpenses
        expenses: selectExpenses(state.expenses, state.filters),
        //  filters:state.filters
        /*
        expenses: state.expenses, //pegando dados da loja 
        filters:state.filters
        */
    };
}
export default connect(mapStateToProps)(ExpenseList);
