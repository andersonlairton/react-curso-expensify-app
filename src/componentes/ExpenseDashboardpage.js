import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilter from './ExpenseListFilter';
import ExpenseSumary from './ExpenseSumary';

const ExpenseDashboardpage=()=>(
    <div>
        este Ã© meu componente dashboard
        <ExpenseSumary/>
        <ExpenseListFilter/>
        <ExpenseList/>
    </div>
);
export default ExpenseDashboardpage