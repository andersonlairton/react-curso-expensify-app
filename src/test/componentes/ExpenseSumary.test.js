import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSumary} from '../../componentes/ExpenseSumary';

test('teste com apenas um item ',()=>
{
    const wrapper = shallow(<ExpensesSumary ExpenseContador={1} ExpenseTotal={235} />);
    expect(wrapper).toMatchSnapshot();
});

test('teste com varias despesas ',()=>
{
    const wrapper = shallow(<ExpensesSumary ExpenseContador={23} ExpenseTotal={5325729878752} />);
    expect(wrapper).toMatchSnapshot();
});
