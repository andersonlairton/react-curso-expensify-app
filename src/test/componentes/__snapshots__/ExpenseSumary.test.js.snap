// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`teste com apenas um item  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpensesSumary
    ExpenseContador={1}
    ExpenseTotal={235}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        Veja o valor total das 
        1
         
        expense
         totais ,com o valor de 
        $2.35
</h1>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Veja o valor total das ",
          1,
          " ",
          "expense",
          " totais ,com o valor de ",
          "$2.35",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Veja o valor total das ",
        1,
        " ",
        "expense",
        " totais ,com o valor de ",
        "$2.35",
      ],
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          Veja o valor total das 
          1
           
          expense
           totais ,com o valor de 
          $2.35
</h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Veja o valor total das ",
            1,
            " ",
            "expense",
            " totais ,com o valor de ",
            "$2.35",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Veja o valor total das ",
          1,
          " ",
          "expense",
          " totais ,com o valor de ",
          "$2.35",
        ],
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`teste com varias despesas  1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ExpensesSumary
    ExpenseContador={23}
    ExpenseTotal={5325729878752}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        Veja o valor total das 
        23
         
        expenses
         totais ,com o valor de 
        $53,257,298,787.52
</h1>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Veja o valor total das ",
          23,
          " ",
          "expenses",
          " totais ,com o valor de ",
          "$53,257,298,787.52",
        ],
      },
      "ref": null,
      "rendered": Array [
        "Veja o valor total das ",
        23,
        " ",
        "expenses",
        " totais ,com o valor de ",
        "$53,257,298,787.52",
      ],
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          Veja o valor total das 
          23
           
          expenses
           totais ,com o valor de 
          $53,257,298,787.52
</h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Veja o valor total das ",
            23,
            " ",
            "expenses",
            " totais ,com o valor de ",
            "$53,257,298,787.52",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Veja o valor total das ",
          23,
          " ",
          "expenses",
          " totais ,com o valor de ",
          "$53,257,298,787.52",
        ],
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
